syntax = "proto3";
option cc_enable_arenas = true;

package NFq.NConfig;
option java_package = "ru.yandex.kikimr.proto";

////////////////////////////////////////////////////////////

message TYdbStorageConfig {
    string Endpoint = 1;
    string Database = 2;
    string OAuthFile = 3;
    string TablePrefix = 4;
    string CertificateFile = 6;
    string IamEndpoint = 7;
    string SaKeyFile = 8;
    bool UseLocalMetadataService = 9;
    uint32 ClientTimeoutSec = 10;
    uint32 OperationTimeoutSec = 11;
    uint32 CancelAfterSec = 12;
    bool UseSsl = 13;
    uint32 TableClientMaxActiveSessions = 14;

    // alternative to OAuthFile to simplify tests
    string Token = 5;
}
